starlette/responses.py

```diff
@@ -290,9 +290,11 @@ def __init__(
         stat_result: typing.Optional[os.stat_result] = None,
         method: typing.Optional[str] = None,
         content_disposition_type: str = "attachment",
+        range: typing.Optional[typing.Tuple[int, int]] = None,
     ) -> None:
         self.path = path
-        self.status_code = status_code
+        self.status_code = status_code if range is None else 206
```

Comment:
The status code is ignored in case of range being sent.

This doesn't look cool.

Reply:
Can you change `status_code: int = 200` to `status_code: Optional[int] = None` then here set the default to 200 if there is no range and 206 if there is a range? If it's explicitly set to something other than 206 and there is a range then throw an error.

Reply:
This would make the `FileResponse` status_code type differ from the other Response classes.

But indeed, I can do the following instead:

```python
        self.status_code = status_code
        self.range = range
        if self.range is not None and self.status_code != 206:
            raise RuntimeError("Range requests must have a 206 status code.")
```

Reply:
Well the `Response.status_code` type would always be an integer still, the default value is just inferred and `None` is used in to signal that it was not passed explicitly.

Throwing an exception seems much better than ignoring the code though :)